src/librustc_middle/query/mod.rs

```diff
@@ -1309,6 +1309,13 @@ rustc_queries! {
         query codegen_unit(_: Symbol) -> &'tcx CodegenUnit<'tcx> {
             desc { "codegen_unit" }
         }
+        query unused_generic_params(key: DefId) -> u64 {
```

Comment:
If we're not going back to a regular set, can you make a `SmallBitSet` type and put it in `rustc_data_structures`?
I wonder if @nnethercote has opinions about this.
Specifically this isn't necessary "small" in the number of elements, but in the domain size (as we suspect more than 64 generic parameters would be rare - we could also bump this to `u128` for little cost).

Reply:
I've not made this change yet.

Reply:
A new `SmallBitSet` type seems reasonable to me. Note that the existing bitsets are in `librustc_index/bit_set.rs`, not `librustc_data_structures/`.

Reply:
Hmm maybe `FiniteBitSet` would make more sense (kind of like `ArrayVec`), maybe even `FiniteBitSet<u64>`?
(you can skip making it fully generic for now and just have `impl FiniteBitSet<u64> {...}` for the methods)

Reply:
Fixed by adding a `FiniteBitSet`.